fetch.bs

```diff
@@ -4405,6 +4523,8 @@ steps:
 
  <li><p><a>Upgrade <var>request</var> to a potentially trustworthy URL, if appropriate</a>.
 
+ <li><p><a href="#https-upgrades-upgrade">Optionally, run HTTPS upgrading algorithm on <var>request</var>, if appropriate</a>.
```

Comment:
Why "optionally" here and below? Also, why do we need to run this algorithm both as part of "main fetch" and "http fetch"?

Reply:
Removed "optionally" below. As I understand, Http fetch is where the redirect can be properly handled.

So the upgrade is done in main fetch and the fallback is handled in http fetch.

Reply:
It's still not clear to me why we need to upgrade in "main fetch", given that any "http" scheme request would go through HTTP Fetch. Can you clarify?

Reply:
This is so that we can still do mixed content upgrades on the upgraded URL (step 6 in the current Main Fetch, which will be right after this line). Mixed content upgrades happen before scheme fetch (step 12) and thus http fetch.

Reply:
OK, cool. Maybe add a note to that effect?

Reply:
Done, PTAL at the wording?